### PR TITLE
Charlesmchen/attachment approval in input toolbar

### PR DESCRIPTION
**Pull request by charlesmchen**
This PR is ready for review. 

PTAL @michaelkirk

### CODE REVIEW

File: Signal/src/ViewControllers/MediaMessageView.swift [hunk 1]
```diff
@@ -355,16 +353,16 @@ class MediaMessageView: UIView, OWSAudioAttachmentPlayerDelegate {
...
+        audioPlayButton?.setImage(image, for: .normal)
         audioPlayButton?.imageView?.tintColor = UIColor.ows_materialBlue()
     }
 
     private func setAudioIconToPause() {
-        let image = UIImage(named:"audio_pause_black_large")?.withRenderingMode(.alwaysTemplate)
+        let image = UIImage(named: "audio_pause_black_large")?.withRenderingMode(.alwaysTemplate)
         assert(image != nil)
-        audioPlayButton?.setImage(image, for:.normal)
+        audioPlayButton?.setImage(image, for: .normal)
```
Comment (charlesmchen) [inline Signal/src/ViewControllers/MediaMessageView.swift]: I'm sorry but `clang-format` behaves differently on my laptop and desktop.

**Comment by charlesmchen** [comment]
screen shot 2017-10-18 at 10 31 53 am

The design is strictly temporary and pending discussion.  My actual proposal is to behave more like iMessage where the "proposed" attachment is on the left, you see a normal "send" button on the right (like a text message) and a there's a "X" to cancel in the upper right of the "proposed" attachment.

**Comment by charlesmchen** [comment]
This PR is ready for review. 

PTAL @michaelkirk

File: Signal/src/ViewControllers/MediaMessageView.swift (hunk 1 above)
Comment (michaelkirk) [reply Signal/src/ViewControllers/MediaMessageView.swift]: This is the correct one =)

File: Signal/src/ViewControllers/ConversationView/ConversationInputToolbar.m [hunk 2]
```diff
@@ -191,6 +207,33 @@ - (void)ensureContentConstraints
 {
     [NSLayoutConstraint deactivateConstraints:self.contentContraints];
 
+    if (self.attachmentToApprove) {
+        self.contentView.hidden = YES;
+        self.attachmentApprovalView.hidden = NO;
+
+        self.contentContraints = @[
+            [self.attachmentApprovalView autoSetDimension:ALDimensionHeight toSize:300.f],
```
Comment (michaelkirk) [inline Signal/src/ViewControllers/ConversationView/ConversationInputToolbar.m]: just checking: 300 is good on iPhone5 and iPhone7+?

File: Signal/src/ViewControllers/ConversationView/ConversationInputToolbar.m [hunk 3]
```diff
@@ -645,6 +688,99 @@ - (void)observeValueForKeyPath:(nullable NSString *)keyPath
...
+
+    UIView *sendButton =
+        [self createAttachmentApprovalButton:NSLocalizedString(
+                                                 @"ATTACHMENT_APPROVAL_SEND_BUTTON", comment
+                                                 : @"Label for 'send' button in the 'attachment approval' dialog.")
+                                       color:[UIColor colorWithRGBHex:0x2ecc71]
+                                    selector:@selector(attachmentApprovalSendPressed)];
+    [buttonRow addSubview:sendButton];
+    [sendButton autoPinHeightToSuperview];
+    [sendButton autoPinEdge:ALEdgeLeft toEdge:ALEdgeRight ofView:buttonSpacer];
```
Comment (michaelkirk) [inline Signal/src/ViewControllers/ConversationView/ConversationInputToolbar.m]: Are we missing a `[sendButton autoPinEdge:ALEdgeRight toEdge:??? ofView:???];`?

File: Signal/src/ViewControllers/ConversationView/ConversationInputToolbar.m (hunk 2 above)
Comment (charlesmchen) [reply Signal/src/ViewControllers/ConversationView/ConversationInputToolbar.m]: Good q - it's part of the design discussion I'm hoping to have in iOS standup once our retreat is over.

File: Signal/src/ViewControllers/ConversationView/ConversationInputToolbar.m (hunk 3 above)
Comment (charlesmchen) [reply Signal/src/ViewControllers/ConversationView/ConversationInputToolbar.m]: With this design, cancel and send buttons are fully constrained, but their size is set in `createAttachmentApprovalButton`.

Comment (charlesmchen) [reply Signal/src/ViewControllers/ConversationView/ConversationInputToolbar.m]: But I expect this design to change quite a bit.

**Pull request merged**